### PR TITLE
Allow to work with semantic versioning

### PR DESCRIPTION
**Pull request by huer12**
The VS extension is not able to work with preview versions. If you have a version like 0.27.3-beta, then the extension does not work. This code is fixing this issue. It allows to work with preview versions.

### CODE REVIEW

**Comment by belav** [comment]
The extension installs the version of csharpier that it detects into a custom location. There have been no releases of csharpier on nuget that use preview number versioning. If someone were to reference a beta version like this wouldn't they also need a nuget source to tell VS where to find said version?

**Comment by huer12** [comment]
We have released such a nuget version for ourself on our nuget server. So you are right VS needs to find it, but it is able to do so in our environment. We thought that may be others are doing this too and decided to bring a fix into the master.

**Comment by shocklateboy92** [comment]
It looks like you still have formatting errors.

**Review by shocklateboy92 — APPROVED** [review]

File: Src/CSharpier.VisualStudio/CSharpier.VisualStudioShared/CSharpierProcessProvider.cs [hunk 1]
```diff
@@ -278,6 +277,27 @@ private ICSharpierProcess SetupCSharpierProcess(string directory, string version
...
+        {
+            if (Version.TryParse(version, out var installedVersion) && installedVersion != null)
+            {
+                return installedVersion;
+            }
+
+            // handle semver versions
+            // regex from https://semver.org/#is-there-a-suggested-regular-expression-regex-to-check-a-semver-string
+            var regex = Regex.Match(
+                "0.27.3-dg.13",
```
Comment (parched) [inline Src/CSharpier.VisualStudio/CSharpier.VisualStudioShared/CSharpierProcessProvider.cs:290]: Shouldn't this be `version`?

Comment (martindisch) [reply Src/CSharpier.VisualStudio/CSharpier.VisualStudioShared/CSharpierProcessProvider.cs:290]: You're right, fixed it in #1264.

**Pull request merged**